src/pages/EnablePayments/AddBankAccount/substeps/Plaid.tsx

```diff
@@ -0,0 +1,112 @@
+import {useIsFocused} from '@react-navigation/native';
+import React, {useCallback, useEffect} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import AddPlaidBankAccount from '@components/AddPlaidBankAccount';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormInputErrors, FormOnyxValues} from '@components/Form/types';
+import useLocalize from '@hooks/useLocalize';
+import type {SubStepProps} from '@hooks/useSubStep/types';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as BankAccounts from '@userActions/BankAccounts';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PersonalBankAccountForm} from '@src/types/form';
+import INPUT_IDS from '@src/types/form/PersonalBankAccountForm';
+import type {PlaidData} from '@src/types/onyx';
+
+type PlaidOnyxProps = {
+    /** The draft values of the bank account being setup */
+    personalBankAccountDraft: OnyxEntry<PersonalBankAccountForm>;
+
+    /** Contains plaid data */
+    plaidData: OnyxEntry<PlaidData>;
+};
+
+type PlaidProps = PlaidOnyxProps & SubStepProps;
+
+const BANK_INFO_STEP_KEYS = INPUT_IDS.BANK_INFO_STEP;
+
+const validate = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.PERSONAL_BANK_ACCOUNT_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.PERSONAL_BANK_ACCOUNT_FORM> => {
+    const errorFields: FormInputErrors<typeof ONYXKEYS.FORMS.PERSONAL_BANK_ACCOUNT_FORM> = {};
+
+    if (!values.selectedPlaidAccountID) {
+        errorFields.selectedPlaidAccountID = 'bankAccount.error.youNeedToSelectAnOption';
+    }
+
+    return errorFields;
+};
+
+function Plaid({personalBankAccountDraft, onNext, plaidData}: PlaidProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const isFocused = useIsFocused();
+    const selectedPlaidAccountID = personalBankAccountDraft?.[BANK_INFO_STEP_KEYS.PLAID_ACCOUNT_ID] ?? '';
+
+    useEffect(() => {
```

Comment:
worth to keep useEffect at the bottom , to have clear variables/ logic section and side effects section